### DISCUSSION TITLE
How to add names to Configuration Objects

### DISCUSSION THREAD

**Discussion by Byron2016** [eslint/eslint#19001] [Q&A] [answered]
Documentation said to use **name key** [Configuration Objects names](https://eslint.org/docs/latest/use/configure/configuration-files#configuration-objects)  but how to add in this case?:

```js
import globals from "globals";
import pluginJs from "@eslint/js";
import markdown from "@eslint/markdown";
// ....
export default [
  pluginJs.configs.recommended,
  ...markdown.configs.recommended,
];
```

![image](https://github.com/user-attachments/assets/689bc22c-26a4-42b9-96af-77951ed285fe)

**Comment by mdjermanovic** [comment, marked as answer]
`@eslint/markdown`'s configs have the `name` property.

`@eslint/js`'s configs don't because those configs are also used in the eslintrc config system, which doesn't accept `name` property. (we previously added names in https://github.com/eslint/eslint/commit/4d11e567baff575146fd267b3765ab2c788aa1e5, but had to revert the change in https://github.com/eslint/eslint/commit/8d189586d60f9beda7be8cdefd4156c023c4fdde). You can add it manually in your config file:

```js
import globals from "globals";
import pluginJs from "@eslint/js";
import markdown from "@eslint/markdown";
// ....
export default [
  {
    name: "js/recommended",
    ...pluginJs.configs.recommended,
  },  
  ...markdown.configs.recommended,
];
```

The rest are, I believe, config objects from the [default configuration](https://github.com/eslint/eslint/blob/main/lib/config/default-config.js) that always automatically applies. If you open a change request, we can consider adding names to these config objects.
